### ISSUE TITLE
Renaming Servers on remote nodes breaks server connection

### ISSUE THREAD

**Issue by Krambed** [PufferPanel/PufferPanel#1375]
**Describe the bug**
Using the panel to rename a server that is on a node other than the localnode seems to break the server, behaving as if the server does not exist and getting 404 when accessing api links or interacting with the server. This issue does not become fixed when altering the server name back to the original in the panel or in the database (needs to be tested). The only solution I have found is to manually remove the server from the panel database and re-add it using the panel as attempting to delete the server from the panel leads to a 404.

**To Reproduce**
1. Deploy a node on another server and add it to the panel.
2. Create any server with any configuration and name on the new node.
3. Rename the server.
4. Attempt to view or edit the server. The server will be unable to query and all api links leads to 404 e.g. https://{DOMAIN}/api/servers/{IDENTIFIER}/query

**Expected behavior**
The server works as expected and all actions on the server such as file transfer and restarting works.

**Node (Panel):**
 - OS: Ubuntu 22.04 aarch64
 - Pufferpanel Version: [v3.0.0-rc.5](https://github.com/pufferpanel/pufferpanel/releases/tag/v3.0.0-rc.5)
 - Database: SQLite

**Node (Remote)**
 - OS: Ubuntu 24.04 aarch64
 - Pufferpanel Version: [v3.0.0-rc.5](https://github.com/pufferpanel/pufferpanel/releases/tag/v3.0.0-rc.5)

**Comment by LordRalex** [comment]
Hm... that should not change the id... which is how the API is based.

If you do a rename, can you confirm that the node still has the JSON file located in /var/lib/pufferpanel/servers and that it contains something?

**Comment by Bluemethyst** [comment]
im having the same error message but not after renaming a server, after adding a node and creating a server on that new second node

**Comment by Krambed** [comment]
In testing of the issue, I got the same result without renaming the server by repeating @Bluemethyst's steps on an "almost" empty template.
![image](https://github.com/user-attachments/assets/704f04e3-c684-47ed-bd13-4dc9ae2c9fc3)
Here is the contents of the json of that server on the remote node. [d4309030.json](https://github.com/user-attachments/files/17839626/d4309030.json)
Upon opening the database file, I notice that the broken server has a missing node_id. Server d4309030 is the broken node previously mentioned while Server 5d02959a currently works as expected. Server 5d02959a was created using the Minecraft Java Template.
![image](https://github.com/user-attachments/assets/5251f3cf-5d44-4c63-8de3-2ec76ec8ab5c)
I then tested adding 1 to the node_id on d4309030, which should correspond to the remote node. Sure enough, it works and I'm able to edit that server without issue, a temporary solution to Bluemethyst's problem until the issue gets fixed. I'll will try to debug my original issue but when I was checking JSON files they did exist on the remote node and still had the identifier so my guess is that this is an issue with the database.
